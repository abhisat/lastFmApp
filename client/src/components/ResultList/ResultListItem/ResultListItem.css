.listItem {
  height: 8rem;
  display: flex;
  cursor: pointer;
}

.itemContent {
  padding: 20px;
}

.itemImage {
  width: 7rem;
  height: 7rem;
}

.resultItemContainer {
  width: 100%;
  display: flex;
}

.itemHeader.itemHeader.itemHeader.itemHeader.itemHeader {
  color: #b90000 !important;
}

.text {
  padding: 2px;
}
