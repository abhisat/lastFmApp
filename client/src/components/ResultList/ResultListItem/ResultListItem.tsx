import React, { SyntheticEvent } from "react";
import { Image, Item, Icon, Container } from "semantic-ui-react";
import { Facebook } from "react-content-loader";
import "./ResultListItem.css";

export interface ResultListItemData {
  name?: string;
  playcount?: string;
  listeners?: string;
  url?: string;
  image?: string;
  isLoading: boolean;
  handleClick?(value: string, e: SyntheticEvent): void;
}

export interface ResultListItemProps extends ResultListItemData {
  isLoading: boolean;
}

const ResultListItem: React.FunctionComponent<ResultListItemProps> = props => {
  const {
    name,
    playcount,
    listeners,
    url,
    image,
    isLoading,
    handleClick
  } = props;

  const listItemClick = (value: string, e: SyntheticEvent) => {
    if (handleClick) {
      handleClick(value, e);
    }
  };

  return (
    <React.Fragment>
      <Container className={"resultItemContainer"}>
        {isLoading ? (
          <Facebook className={"loader"} />
        ) : (
          <Item className={"listItem"} onClick={e => listItemClick(name, e)}>
            <Image className={"itemImage"} src={image} />
            <Item.Content className={"itemContent"}>
              <Item.Header as='a' className={"itemHeader text"}>
                {name}
              </Item.Header>
              <Item.Meta className={"text"}>
                <Icon name='play' />
                {playcount}
              </Item.Meta>
              <Item.Extra className={"text"}>
                <Icon name='users' />
                {listeners}
              </Item.Extra>
              <Item.Extra className={"itemTags text"}>
                <Icon name='linkify' />
                {url}
              </Item.Extra>
            </Item.Content>
          </Item>
        )}
      </Container>
    </React.Fragment>
  );
};

export { ResultListItem };
