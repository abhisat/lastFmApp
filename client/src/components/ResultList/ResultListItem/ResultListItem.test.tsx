import React from "react";
import { ResultListItem, ResultListItemProps } from "./ResultListItem";
import EnzymeAdapter from "enzyme-adapter-react-16";
import Enzyme, { mount } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({
  adapter: new EnzymeAdapter()
});

const handleClick: jest.Mock = jest.fn();

const props: ResultListItemProps = {
  name: "Arctic Monkeys",
  playcount: "349443530",
  listeners: "3618819",
  isLoading: false,
  url: "https://www.last.fm/music/Arctic+Monkeys",
  image:
    "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
  handleClick: handleClick
};

const mountApp = () => {
  const wrapper: Enzyme.ReactWrapper = mount(<ResultListItem {...props} />);
  wrapper.update();
  return wrapper;
};

describe("<ArtistListItem/>", () => {
  let App: Enzyme.ReactWrapper;

  beforeEach(() => {
    App = mountApp();
  });

  afterEach(() => {});

  it("renders without crashing", () => {
    mountApp();
  });

  it("Matches the snapshot", () => {
    expect(toJson(App)).toMatchSnapshot();
  });
});
