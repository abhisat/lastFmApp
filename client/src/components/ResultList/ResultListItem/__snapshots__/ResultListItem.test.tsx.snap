// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtistListItem/> Matches the snapshot 1`] = `
<ResultListItem
  handleClick={[MockFunction]}
  image="https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png"
  isLoading={false}
  listeners="3618819"
  name="Arctic Monkeys"
  playcount="349443530"
  url="https://www.last.fm/music/Arctic+Monkeys"
>
  <Container
    className="resultItemContainer"
  >
    <div
      className="ui container resultItemContainer"
    >
      <Item
        className="listItem"
        onClick={[Function]}
      >
        <div
          className="item listItem"
          onClick={[Function]}
        >
          <Image
            as="img"
            className="itemImage"
            src="https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png"
            ui={true}
          >
            <img
              className="ui image itemImage"
              src="https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png"
            />
          </Image>
          <ItemContent
            className="itemContent"
          >
            <div
              className="content itemContent"
            >
              <ItemHeader
                as="a"
                className="itemHeader text"
              >
                <a
                  className="header itemHeader text"
                >
                  Arctic Monkeys
                </a>
              </ItemHeader>
              <ItemMeta
                className="text"
              >
                <div
                  className="meta text"
                >
                  <Icon
                    as="i"
                    name="play"
                  >
                    <i
                      aria-hidden="true"
                      className="play icon"
                      onClick={[Function]}
                    />
                  </Icon>
                  349443530
                </div>
              </ItemMeta>
              <ItemExtra
                className="text"
              >
                <div
                  className="extra text"
                >
                  <Icon
                    as="i"
                    name="users"
                  >
                    <i
                      aria-hidden="true"
                      className="users icon"
                      onClick={[Function]}
                    />
                  </Icon>
                  3618819
                </div>
              </ItemExtra>
              <ItemExtra
                className="itemTags text"
              >
                <div
                  className="extra itemTags text"
                >
                  <Icon
                    as="i"
                    name="linkify"
                  >
                    <i
                      aria-hidden="true"
                      className="linkify icon"
                      onClick={[Function]}
                    />
                  </Icon>
                  https://www.last.fm/music/Arctic+Monkeys
                </div>
              </ItemExtra>
            </div>
          </ItemContent>
        </div>
      </Item>
    </div>
  </Container>
</ResultListItem>
`;
